src/main/java/com/faforever/client/teammatchmaking/MatchmakingQueueItemController.java

```diff
@@ -140,17 +145,19 @@ private void setQueuePopTimeUpdater(MatchmakerQueueBean queue) {
   }
 
   public void onJoinLeaveQueueClicked() {
-    if (queueJoined.getValue()) {
-      teamMatchmakingService.leaveQueue(getQueue());
-    } else {
-      teamMatchmakingService.joinQueue(getQueue()).thenAccept(success -> {
-        if (!success) {
-          joinLeaveQueueButton.setSelected(false);
-          refreshingLabel.setVisible(false);
-        }
-      });
+    if (teamMatchmakingController.searchButton.isSelected()) {
```

Comment:
This should be a property of the teamMatchmakingService not derived solely from the controller